### DISCUSSION TITLE
Cannot import the DAG file with _config suffix.

### DISCUSSION THREAD

**Discussion by yc0** [apache/airflow#43575] [Q&A] [answered]
### Apache Airflow version

2.10.2

### What happened?

The  DAG file name goes with ``_config.py``, and the DAG won't be imported. For instance, the DAG is named xxxx_config.py, it won't be imported successfully. 

### How to reproduce

create a DAG, and name as wf_sample_config.py. 
the context in wf_sample_config.py is as follows:
```python
from datetime import datetime
from airflow.decorators import  task
from airflow import DAG

with DAG (
    dag_id="a_testing_for_naming",
    start_date=datetime(2024, 10, 31),
    schedule=None
) as dag:

    @task
    def start(params = None, task=None, ti=None, dag_run=None):
        pass

    start()
```

You won't see the a_testing_for_naming DAG in your DAG view in the web UI.
How do you solve it?
just change the name with _cfg. For example, wf_sample_cfg.py. You can see the DAG loaded successfully.

### Operating System

Debian GNU/Linux 11 (bullseye)

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

python version : python 3.12

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by rawwar** [comment]
Before I test this out, I have a quick question: Can you please check your .airflowignore for any entries that might be causing this?

EDIT: I just tested this on 2.10.2, and I can't replicate this issue.

**Comment by yc0** [comment]
here comes our .airflowignore
```
.archive/*
.config/*
.lib/*
extends/*
```

**Comment by rawwar** [comment, marked as answer]
so, by default [dag_ignore_file_syntax](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#dag-ignore-file-syntax) is set to regex. And, it looks like `.config/*` is getting matched with any file with suffix config`. I tried setting `AIRFLOW__CORE__DAG_IGNORE_FILE_SYNTAX=glob` and it did not ignore my dag(file named `example_config.py`)


Just tested. 
```
In [2]: re2.findall(r".config/*", "example_config.py")
Out[2]: ['_config']
```

Solution for you is to use glob syntax and set AIRFLOW__CORE__DAG_IGNORE_FILE_SYNTAX=glob
